convert2rhel/toolopts.py

```diff
@@ -78,6 +75,29 @@ def set_opts(self, supported_opts):
             if value is not None and hasattr(self, key):
                 setattr(self, key, value)
 
+    @property
+    def should_subscribe(self):
```

Comment:
This feels to me that it shouldn't belong here... This feels that it should be placed in the subscription.py module, but, we are using this same property in another part of the toolopts as well, so I'm not sure how to feel about it. 

I see how useful the property is here, but still, it feels to be that it should belong to subscription.py, as we are not really taking the CLI option and parsing it to assign with the global toolopts object, but rather, we are saying "take those already defined CLI options and made a decision with it". 

Any thoughts? I'm cool to leave the property here, but still, I think I should bring this comment just to make sure there is nothing we can do to move it to a "more appropriate module".

Reply:
I have the same uncertainty as you here.  I think asking the question "Should I subscribe" does feel like it should be answered by subscription.py.  OTOH, I think that toolopts.py should definitely not `import subscription`.  So I think that leaves us with these choices:

* Just have the code here and public.
* Put code which does this check in both toolopts.py (private, just for validating the other arguments) and subscription.py (for public consumption by other code).
* Put the code into a "utils" module.  Note that this would be a separate file from the current utils so that utils can keep the stdlib as its only dependency.  (Maybe making utils into a directory and making a subscription module inside of that.)  I started down this path so I know it's workable; I backed it out because it is more intrusive 

The first of these is what is currently implemented.  The last one is my preference if we think this belongs somewhere else.

What do you think?

Reply:
I have moved this although I went with a variation on solution 2 that doesn't duplicate the body of the code.    I implemented should_subscribe as a private function in toolopts.py so that it can be used there.  Then I created a public identifier for should_subscribe in subscription.py which just references the private function with appropriate arguments.

The reason I decided not to do utils isn't technical but maintainability.  I couldn't figure out a good way to show people that `utils.py` only requires the stdlib but `subscription_utils.py` would require `convert2rhel.toolopts`.  If I named it `toolopts_utils.py` to signify that it required `toolopts`, then we would have the same feeling that it belonged in something related to subscriptions, not in something related to toolopts.